### DISCUSSION TITLE
Mensagem padrão de validação do Enum

### DISCUSSION THREAD

**Discussion by RanieryCG** [balta-io/2811#26] [Q&A] [answered]
Uma coisa que eu não consegui fazendo o model binding foi mudar a mensagem padrão "The value {0} is not valid for {1}" de um atributo do tipo enum.

Como faço isso? 

Mudando o error message do required não funciona e nem utilizando o EnumDataType. 

Nem criando um ValidationAttribute.

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @RanieryCG, como vai?

Não me recordo muito bem a aula que falei sobre isto, qual seria?
Em adicional, se tiver algum trecho de código para ajudar...

**Reply by RanieryCG** [reply]
Oi @andrebaltieri, você não falou explicitamente em nenhum curso, eu só estava fazendo os fundamentos do ASP.NET 6 e fui usar um Enum como tipo de atributo no request e percebi isso.

Exemplo, na validação abaixo, se eu mandar um valor fora do que o enum recebe ou mandar zero, ele não cai no required e nem na mensagem de erro do EnumDataType. 

Se eu mandar um valor que não consta no enum (imagem 2), exemplo -123, ele retorna a mensagem "The value '-123' is invalid." Esta mensagem não estou conseguindo sobreescrever de jeito nenhum.

![image](https://user-images.githubusercontent.com/6059830/177182686-01d9c04d-3aff-445f-ae43-c6468ab851a5.png)

![image](https://user-images.githubusercontent.com/6059830/177183019-d1119940-b17c-44ea-9287-2299c24d5d36.png)

**Reply by andrebaltieri** [reply]
@RanieryCG acredito que este erro não é do Enum em si... acredito que nem chegou nele na verdade. Você consegue colocar um valor positivo (Tipo 220) para testar se a mensagem aparece?

Se aparecer, aí você trata os negativos de outra forma, com o MinLen por exemplo...

**Reply by RanieryCG** [reply]
@andrebaltieri, sim. Com o valor maior a mesma coisa. Eu só queria trocar essa mensagem. "The value {0} is invalid".

**Reply by andrebaltieri** [reply]
Neste caso, talvez tenha que criar um validador customizado. Inclusive parece até um bug no .net esse comportamento :)
